### PR TITLE
Fix #765: Remove gx

### LINKED ISSUE #765: Fully remove Gx
They don't bother anyone but probably will never be used again
This would mean getting rid off `deptools` as well

### PR DESCRIPTION
**Pull request by pkafei**
Deleted `gx` from ipfs-cluster code base.

### CODE REVIEW

**Comment by pkafei** [comment]
@kishansagathiya Can you tell me specifically which important targets should be kept in MAKEfile? It also seems like I need to also figure out why the build is failing.

**Comment by hsanjuan** [comment]
@pkafei simply don't delete any targets that are not gx-specific. For example, the `install` target. Some tests are failing because of that.

**Comment by kishansagathiya** [comment]
@pkafei The commits that you have made are from the user `Protocol Labs` instead of `pkafei`, I don't know if that can be a problem, but It's better if you change that.

**Comment by pkafei** [comment]
Also will run tests before I push up code again!

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,5 @@
 deptools=deptools
 sharness = sharness/lib/sharness
-GXENABLED=no # Set to yes for Gx builds.
 gx=$(deptools)/gx
 gx-go=$(deptools)/gx-go
```
Comment (kishansagathiya) [inline Makefile]: gx and gx-go should be unused now

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,5 @@
 deptools=deptools
```
Comment (kishansagathiya) [inline Makefile]: unused, this was used only to store binaries of gx and gx-go

File: Makefile [hunk 3]
```diff
@@ -34,32 +33,6 @@ service: gx-deps
 ctl: gx-deps
```
Comment (kishansagathiya) [inline Makefile]: Don't use removed targets `docker_gx-deps`, `gx-deps` (here and at all other places)

File: Makefile [hunk 4]
```diff
@@ -121,4 +82,4 @@ docker-compose:
 
 prcheck: gx-deps check service ctl test
 
-.PHONY: all gx gx-deps test test_sharness clean_sharness rw rwundo publish service ctl install clean gx-clean docker
+.PHONY: all gx gx-deps test test_sharness clean_sharness rw rwundo publish service ctl install clean gx-clean docker
```
Comment (lanzafame) [inline Makefile]: Remove `gx gx-deps gx-clean`

**Comment by lanzafame** [comment]
The go module env to expose is `GO111MODULE=one`

**Comment by kishansagathiya** [comment]
I have added changes to clear doubts arising here (https://github.com/ipfs/ipfs-cluster/issues/825), hope that's okay

File: Makefile [hunk 5]
```diff
@@ -1,49 +1,44 @@
...
 	@rm -rf ./test/testingData
 	@rm -rf ./compose
 
-install: gx-deps
+install:
 	$(MAKE) -C cmd/ipfs-cluster-service install
 	$(MAKE) -C cmd/ipfs-cluster-ctl install
 
-docker_install: docker_gx-deps
+docker_install:
```
Comment (hsanjuan) [inline Makefile]: the `docker` target is missing

**Comment by hsanjuan** [comment]
Needs rebase, also let's make sure `GO111Module=on` is exported on the `Makefile`. Can you finish this @kishansagathiya ?

File: Makefile (hunk 5 above)
Comment (kishansagathiya) [reply Makefile]: `docker_install` was install but gx would use local deps . Now with `gx`gone, `docker_install` is same as `install`, so probably we can get rid off this target.

**Comment by lanzafame** [comment]
@kishansagathiya can we put the badger fixes in a separate PR?

**Comment by kishansagathiya** [comment]
@lanzafame 
> @kishansagathiya can we put the badger fixes in a separate PR?

We can always put that in a separate commit to show it differently (and remove gx in a separate commit). If that's not good, I can surely make a new PR.

**Comment by hsanjuan** [comment]
I have removed the deps update. We need to do a more complicated depth bubbling also involving libp2p-core etc so I'll do that in a different PR. I'll merge this one for the moment.

**Pull request merged**